### PR TITLE
[11.0][ADD] theme_flexible

### PR DESCRIPTION
**Pull request by tarteo**
This theme is exceptionally configurable.
You can set the colors, fonts and layout you want (not limited by presets).

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Should we open OCA/website-themes?

Although official names start with `theme_*`, maybe we should use the naming convention `website_theme_*`. What do you think?

**Comment by Fuchur777** [comment]
Open an OCA theme repo.
Keeping with the oca naming convention sounds the way to go

**Comment by tarteo** [comment]
_website_theme__ naming convention sounds good as there can also be _web_theme__ i guess and to prevent conflicting with any other theme.

File: theme_flexible/README.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+Contributors
+------------
+
+* Dennis Sluijk <d.sluijk@onestein.nl>
+* Frank Schellenberg <f.schellenberg@onestein.nl>
+
+Do not contact contributors directly about support or help with technical issues.
+
+Funders
```
Comment (simahawk) [inline theme_flexible/README.rst]: This block should be used to mention external funders. I don't think you need it ;)

File: theme_flexible/models/theme_flexible.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            default='Arial'
+        ))
+        setattr(cls, 'font_%s_google' % name, fields.Boolean())
+        setattr(cls, 'font_%s_weight' % name, fields.Integer(
+            default=400
+        ))
+        setattr(cls, 'font_%s_italic' % name, fields.Boolean())
+        setattr(cls, 'font_%s_underline' % name, fields.Boolean())
+
+    google_query = fields.Char(compute='_compute_google_query')
```
Comment (simahawk) [inline theme_flexible/models/theme_flexible.py]: don't you want to store this?

**Comment by pedrobaeza** [comment]
Yes, that's the case I was thinking in. And congratulations for bringing up the first OCA website theme module!

**Comment by JordiBForgeFlow** [comment]
@rgarnau can you review?

**Review by rgarnau — APPROVED** [review]
LGTM :+1:

**Review by rgarnau — COMMENTED** [review]
I've noticed that when using this theme if you press 'Customize > HTML/CSS Editor' the editor does not open.

**Comment by Fuchur777** [comment]
> I've noticed that when using this theme if you press 'Customize > HTML/CSS Editor' the editor does not open.

You are right. @tarteo

**Comment by tarteo** [comment]
@rgarnau I fixed the bug, can you try again please?

**Comment by astirpe** [comment]
```
FAIL: test_0_0_0_0 (odoo.addons.website_multi_theme.tests.test_assets.UICase)
` Check 0.0.0.0 downloads its default assets.
Traceback (most recent call last):
`   File "/home/travis/build/OCA/website/website_multi_theme/tests/test_assets.py", line 27, in test_0_0_0_0
`     "//head/link[contains(@href, 'web.assets_frontend')]"))
` AssertionError: [] is not true
FAIL: test_localhost (odoo.addons.website_multi_theme.tests.test_assets.UICase)
` Check localhost downloads its multiwebsite-enabled assets.
Traceback (most recent call last):
`   File "/home/travis/build/OCA/website/website_multi_theme/tests/test_assets.py", line 19, in test_localhost
`     'website_multi_theme.auto_assets_website')]"""))
` AssertionError: [] is not true
```

**Comment by rgarnau** [comment]
@tarteo still not working!

**Comment by Fuchur777** [comment]
@tarteo @rgarnau 
Just tested the fix on Chrome / Firefox / Edge.
HTML/CSS editor appears as normal.

image

**Comment by rgarnau** [comment]
@Fuchur777 you're right - just checked again and it works just fine!

**Comment by Fuchur777** [comment]
@astirpe can you review to?

**Review by astirpe — CHANGES_REQUESTED** [review]
Could you fix the lint warnings?
https://travis-ci.org/OCA/website/jobs/342266796

**Comment by astirpe** [comment]
Still some pylint warnings:

```
theme_flexible/static/src/js/theme_customize.js:127: [W7903(javascript-lint), ] 'SelectGoogleFontDialog' was used before it was defined. [Error/no-use-before-define]
theme_flexible/static/src/js/theme_customize.js:222: [W7903(javascript-lint), ] Missing radix parameter. [Error/radix]
theme_flexible/views/res_config_settings_view.xml:1: [W7908(missing-newline-extrafiles), ]  Missing newline
theme_flexible/README.rst:1: [W7908(missing-newline-extrafiles), ]  Missing newline
```

**Review by astirpe — APPROVED** [review]

**Comment by tarteo** [comment]
@pedrobaeza Should I change the name to _website_theme_flexible_. Has it been decided yet?

**Comment by pedrobaeza** [comment]
Yes, please, proceed with the renaming.

**Comment by tarteo** [comment]
@pedrobaeza renamed the module.

**Comment by Fuchur777** [comment]
@pedrobaeza can it be merged?

**Comment by pedrobaeza** [comment]
What do you think about the OCA/website-themes repository creation?

**Comment by tarteo** [comment]
@pedrobaeza Sounds good!

**Comment by Fuchur777** [comment]
Sounds good to. Can you proceed with this @pedrobaeza ?

**Comment by pedrobaeza** [comment]
Repository created, populated and configured in https://github.com/OCA/website-themes

File: website_theme_flexible/templates/options/layout.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+<!-- Copyright 2018 Onestein
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+    <template id="assets_frontend_layout" inherit_id="website.assets_frontend">
+        <xpath expr=".">
+            <style type="text/less">
+                <t t-if="website and website.theme_flexible_id">
+                    <t t-if="website.theme_flexible_id.snippet_border_radius">
+                        #wrap > * {
```
Comment (simahawk) [inline website_theme_flexible/templates/options/layout.xml:11]: watch out w/ `*` selector, they are slow to render

**Review by simahawk — APPROVED** [review]
although there are bits that can be improved this is a good starting point for such feature. tnx! :)

**Comment by simahawk** [comment]
please, squash before merging / moving to another repo

**Comment by pedrobaeza** [comment]
Closing this one for moving it to OCA/website-themes

**Pull request closed without merging**